usr/share/rear/layout/save/GNU/Linux/230_filesystem_layout.sh

```diff
@@ -231,6 +231,13 @@ fi
                 echo -n " uuid=$uuid label=$label"
                 ;;
             (btrfs)
+                # FIXME: Support for multi-disk BTRFS should be implemented sooner or later
+                # because it is the default layout for multi-disk Fedora Workstation installations.
+                # See: https://github.com/rear/rear/issues/2028
+                if ! grep -q "Total devices 1" <(btrfs filesystem show "$mountpoint"); then
```

Comment:
I think this is if condition risky regarding backward compatibility
because it errors out as "fallback" when
```
btrfs filesystem show "$mountpoint"
```
does not contain "Total devices 1".
I don't know how `btrfs filesystem show` behaves
on whatever old and older Linux systems.
I would prefer a condition to error out only
when the actual error condition was detected
but I don't know if it is possible to implement it
this way (with reasonable effort).

Reply:
Also works for me on SLES12 and SLES15 with
```
... grep -q 'Total devices 1 '
```
i.e. with a trailing blank to ensure it won't falsely
match on a btrfs which has for 10 to 19
(or even 100 ... 199 disks ;-)

@lzaoral 
this may look very nitpicking but I had already seen
too many cases in ReaR that worked only for one-digit
numbers so better safe than sorry...

Reply:
> I think this is if condition risky regarding backward compatibility

Good catch!  I can also flip the condition. Something like `grep -qE 'Total devices ([2-9]|[1-9]\d+)'` should do the trick.

Reply:
@lzaoral 
this also works for me on SLES 12 SP5
```
# for mp in $( findmnt -a -n -l -t btrfs -o TARGET ) ; \
  do btrfs filesystem show $mp \
     | grep -qE 'Total devices ([2-9]|[1-9]\d+) ' \
     && echo FAILED for $mp \
     || echo OK for $mp ; \
  done
OK for /
OK for /var/lib/pgsql
OK for /var/lib/mailman
OK for /var/lib/named
OK for /tmp
OK for /opt
OK for /boot/grub2/i386-pc
OK for /var/spool
OK for /var/lib/mariadb
OK for /.snapshots
OK for /var/lib/machines
OK for /var/crash
OK for /srv
OK for /var/lib/mysql
OK for /var/cache
OK for /home
OK for /var/lib/libvirt/images
OK for /var/opt
OK for /var/tmp
OK for /var/log
OK for /boot/grub2/x86_64-efi
OK for /usr/local
```
and on SLES 15 SP6:
```
# for mp in $( findmnt -a -n -l -t btrfs -o TARGET ) ; \
  do btrfs filesystem show $mp \
     | grep -qE 'Total devices ([2-9]|[1-9]\d+) ' \
     && echo FAILED for $mp \
     || echo OK for $mp ; \
  done
OK for /
OK for /.snapshots
OK for /boot/grub2/i386-pc
OK for /boot/grub2/x86_64-efi
OK for /home
OK for /opt
OK for /root
OK for /srv
OK for /tmp
OK for /usr/local
OK for /var
```

Reply:
Off-topic: Ok, I have just shot myself in the foot with the previous suggestion...  I use macOS as my host system (one of the reasons was to be able to test things on aarch64) and I have only tested the previous `grep` on the host and apparently GNU `grep` and FreeBSD `grep` accept different regular expressions with `-E` ...  I've fixed the the previous comment to work with GNU `grep` as well.

Reply:
GNU Grep fails in the original suggestion:
```
$ for i in {1..20}; do echo "Total devices $i " | grep -E 'Total devices ([2-9]|[1-9]\d+) '; done
grep: warning: stray \ before d
grep: warning: stray \ before d
Total devices 2
grep: warning: stray \ before d
Total devices 3
grep: warning: stray \ before d
Total devices 4
grep: warning: stray \ before d
Total devices 5
grep: warning: stray \ before d
Total devices 6
grep: warning: stray \ before d
Total devices 7
grep: warning: stray \ before d
Total devices 8
grep: warning: stray \ before d
Total devices 9
grep: warning: stray \ before d
grep: warning: stray \ before d
grep: warning: stray \ before d
grep: warning: stray \ before d
grep: warning: stray \ before d
grep: warning: stray \ before d
grep: warning: stray \ before d
grep: warning: stray \ before d
grep: warning: stray \ before d
grep: warning: stray \ before d
grep: warning: stray \ before d
```
but the FreeBSD one is fine with that.

Reply:
Also
```
grep -qE 'Total devices ([2-9]|[1-9][0-9]+) '
```
works for me on SLES 12 SP5
```
# for mp in $( findmnt -a -n -l -t btrfs -o TARGET ) ; \
  do btrfs filesystem show $mp \
     | grep -qE 'Total devices ([2-9]|[1-9][0-9]+) ' \
     && echo FAILED for $mp \
     || echo OK for $mp ; \
  done
OK for /
OK for /var/lib/pgsql
OK for /var/lib/mailman
OK for /var/lib/named
OK for /tmp
OK for /opt
OK for /boot/grub2/i386-pc
OK for /var/spool
OK for /var/lib/mariadb
OK for /.snapshots
OK for /var/lib/machines
OK for /var/crash
OK for /srv
OK for /var/lib/mysql
OK for /var/cache
OK for /home
OK for /var/lib/libvirt/images
OK for /var/opt
OK for /var/tmp
OK for /var/log
OK for /boot/grub2/x86_64-efi
OK for /usr/local
```
and on SLES 15 SP6
```
# for mp in $( findmnt -a -n -l -t btrfs -o TARGET ) ; \
  do btrfs filesystem show $mp \
     | grep -qE 'Total devices ([2-9]|[1-9][0-9]+) ' \
     && echo FAILED for $mp \
     || echo OK for $mp ; \
  done
OK for /
OK for /.snapshots
OK for /boot/grub2/i386-pc
OK for /boot/grub2/x86_64-efi
OK for /home
OK for /opt
OK for /root
OK for /srv
OK for /tmp
OK for /usr/local
OK for /var
```

Reply:
Could somebody please share some examples of a multi disk BTRFS and the output of the command?

Reply:
> Could somebody please share some examples of a multi disk BTRFS and the output of the command?

Sure. First example is a single disk fs and the second one is a degraded RAID 1 on three disks with second one missing:

```
$ sudo btrfs filesystem show
Label: 'fedora_fedora'  uuid: 0fa548a6-3544-4f68-b415-ba5e5d9e6e57
        Total devices 1 FS bytes used 41.89GiB
        devid    1 size 126.41GiB used 54.02GiB path /dev/vda3

Label: 'test_fs'  uuid: 77f2aca3-0855-441f-87c8-88e1112b0e44
        Total devices 3 FS bytes used 144.00KiB
        devid    1 size 1.00GiB used 126.38MiB path /dev/loop0
        devid    3 size 1.00GiB used 0.00B path /dev/loop2
        *** Some devices missing
```

Reply:
@jsmeix Addressed in cc151f3f41455f291356495729aca0a5125b8f13.

Reply:
This is contrary to LVM, where lsblk shows the filesystem and VG correctly on all the component disks:
```
# lsblk
NAME                                     MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
vda                                      252:0    0   10G  0 disk 
├─vda1                                   252:1    0    1M  0 part 
├─vda2                                   252:2    0    1G  0 part /boot
└─vda3                                   252:3    0    9G  0 part 
  └─rhel_kvm--08--guest32-pool00_tdata   253:1    0 35.1G  0 lvm  
    └─rhel_kvm--08--guest32-pool00-tpool 253:2    0 35.1G  0 lvm  
      ├─rhel_kvm--08--guest32-root       253:3    0 35.1G  0 lvm  /
      └─rhel_kvm--08--guest32-pool00     253:5    0 35.1G  1 lvm  
vdb                                      252:16   0   10G  0 disk 
├─vdb1                                   252:17   0    8M  0 part 
├─vdb2                                   252:18   0    1G  0 part 
└─vdb3                                   252:19   0    9G  0 part 
vdc                                      252:32   0   10G  0 disk 
└─vdc1                                   252:33   0   10G  0 part 
  └─rhel_kvm--08--guest32-pool00_tdata   253:1    0 35.1G  0 lvm  
    └─rhel_kvm--08--guest32-pool00-tpool 253:2    0 35.1G  0 lvm  
      ├─rhel_kvm--08--guest32-root       253:3    0 35.1G  0 lvm  /
      └─rhel_kvm--08--guest32-pool00     253:5    0 35.1G  1 lvm  
vdd                                      252:48   0   10G  0 disk 
└─vdd1                                   252:49   0   10G  0 part 
  └─rhel_kvm--08--guest32-pool00_tdata   253:1    0 35.1G  0 lvm  
    └─rhel_kvm--08--guest32-pool00-tpool 253:2    0 35.1G  0 lvm  
      ├─rhel_kvm--08--guest32-root       253:3    0 35.1G  0 lvm  /
      └─rhel_kvm--08--guest32-pool00     253:5    0 35.1G  1 lvm  
vde                                      252:64   0   10G  0 disk 
└─vde1                                   252:65   0   10G  0 part 
  └─rhel_kvm--08--guest32-pool00_tdata   253:1    0 35.1G  0 lvm  
    └─rhel_kvm--08--guest32-pool00-tpool 253:2    0 35.1G  0 lvm  
      ├─rhel_kvm--08--guest32-root       253:3    0 35.1G  0 lvm  /
      └─rhel_kvm--08--guest32-pool00     253:5    0 35.1G  1 lvm  
vdf                                      252:80   0   10G  0 disk 
└─vdf1                                   252:81   0   10G  0 part 
  ├─rhel_kvm--08--guest32-pool00_tmeta   253:0    0   36M  0 lvm  
  │ └─rhel_kvm--08--guest32-pool00-tpool 253:2    0 35.1G  0 lvm  
  │   ├─rhel_kvm--08--guest32-root       253:3    0 35.1G  0 lvm  /
  │   └─rhel_kvm--08--guest32-pool00     253:5    0 35.1G  1 lvm  
  └─rhel_kvm--08--guest32-swap           253:4    0  3.9G  0 lvm  [SWAP]
```